### ISSUE TITLE
Refactor variable creation in Web App

### ISSUE THREAD

**Issue by dtasev** [ISISScientificComputing/autoreduce#1042]
Issue raised by: [developer/user/project requirement]

### What?
The whole variable creation code was, at some point, copy pasted into the web app (or the other way around, not sure). Regardless, the duplication needs to go and the two need to be merged into a common version with #1002

This should be done after #1022

**Comment by dtasev** [comment]
Closed by #1042
